azurerm/resource_arm_virtual_network_gateway_connection.go

```diff
@@ -125,6 +125,16 @@ func resourceArmVirtualNetworkGatewayConnection() *schema.Resource {
 				Computed: true,
 			},
 
+			"connection_protocol": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(network.IKEv1),
+					string(network.IKEv2),
+				}, true),
```

Comment:
we're making all new fields case-sensitive going forward - so can we make this:

```suggestion
				}, false),
```

Reply:
Wouldn't this break during conversion to `network.VirtualNetworkGatewayConnectionProtocol` on line 505?

Reply:
It shouldn't as we are validating the strings against the correct case